gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/data/AcceptedLanguageInterceptor.kt

```diff
@@ -0,0 +1,43 @@
+package com.gravatar.quickeditor.data
+
+import android.content.Context
+import androidx.core.app.LocaleManagerCompat
+import androidx.core.os.LocaleListCompat
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import java.util.Locale
+
+internal class AcceptedLanguageInterceptor(private val context: Context) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        return chain.proceed(
+            chain.request().newBuilder()
+                .addAcceptedLanguageHeader(context.languagesList.asLocaleList.acceptedLanguageHeader).build(),
+        )
+    }
+
+    private fun Request.Builder.addAcceptedLanguageHeader(languages: String) = addHeader("Accept-Language", languages)
+
+    private val List<Locale>.acceptedLanguageHeader: String
+        get() {
+            var weight = 1.0F
+            return map { it.language }.reduce { accumulator, language ->
+                weight -= 0.1F
+                "$accumulator,$language;q=$weight"
+            }
```

Comment:
This produces something like:

<img width="517" alt="SCR-20240930-kizm" src="https://github.com/user-attachments/assets/0b919a2e-dcef-41f4-abf2-485b666da9fa">

This is caused by the repeated 0.1F subtraction and how the floating-point works internally.

We can round those values or replace with something like:

```suggestion
            return map { it.language }.mapIndexed { index, language ->
                val weight = 1.0F - (index * 0.1F)
                "$language;q=$weight"
            }.joinToString(",")
```

Whose output is:
<img width="397" alt="SCR-20240930-kkfm" src="https://github.com/user-attachments/assets/06d6031d-9a0f-4c0d-b80e-9409706073f6">

Reply:
Great catch - I've fixed this by using `BigDecimal`. I added some tests as well.